.circleci/config.yml

```diff
@@ -59,30 +69,35 @@ jobs:
             done && [ "$e" -eq "0" ]
 
       - run:
-          name: Docker save
+          name: Docker push to local registry
           no_output_timeout: 40m
           command: |
-            mkdir -p /tmp/cache
-            docker save ubuntu:xenial-20161213 poldracklab/smriprep:latest \
-            | pigz -8 -p 3 > /tmp/cache/docker.tar.gz
+            docker tag poldracklab/smriprep:latest localhost:5000/smriprep
+            docker push localhost:5000/smriprep
+      - run:
+          name: Docker registry garbage collection
+          command: |
+            docker exec -it registry /bin/registry garbage-collect --delete-untagged \
+                /etc/docker/registry/config.yml
 
       - persist_to_workspace:
           root: /tmp
           paths:
-            - cache/docker.tar.gz
+            - docker
             - src/smriprep
 
   update_cache:
-    docker:
-      - image: docker:18.01.0-ce-git
+    machine:
+      # Ubuntu 14.04 with Docker 17.10.0-ce
+      image: circleci/classic:201711-01
     working_directory: /tmp/src/smriprep
     steps:
       - attach_workspace:
           at: /tmp
       - save_cache:
-         key: build-v1-{{ .Branch }}-{{ epoch }}
+         key: build-v2-{{ .Branch }}-{{ epoch }}
          paths:
-            - /tmp/cache/docker.tar.gz
+            - /tmp/docker
```

Comment:
In general I've found caches to be quicker, so there's no advantage to putting things in the workspace and then copying over to the cache in a separate job. Just purely use the cache.

Reply:
See, e.g, https://github.com/poldracklab/fmriprep/pull/1608